### PR TITLE
chore(dev): drop unneeded dependencies and coalesce some others

### PR DESCRIPTION
**Pull request by tobz**
This PR works to reduce some duplicate crate dependencies, and in turn, the need to compile them.

In no particular order:
- switch from `gouth` to `goauth` the `stackdriver_metrics` sink (`stackdriver_logs` already used `goauth`)
- remove `strum`/`strum_macros` as they provide little value over simple hand-rolled impls of common traits, at least where we happened to use them
- _remove_ the `default-features = false` for `syn` in `vector-config-common`, which was manifesting as an entirely distinct version of `syn` needing to be compiled
- switch to `darling@0.13` (down from `0.14`) for `vector-config-common` and friends, as this let us line up with other dependencies we have which use `0.13`

For both clean debug and release builds, this shaves a good 5-10s off the full build. With these changes, the debug binary is about 10MB smaller (but still 2.47GB) and the release binary is about 2MB smaller (but still 148MB).

### CODE REVIEW

File: src/api/schema/components/source.rs [hunk 1]
```diff
@@ -15,7 +13,8 @@ use crate::{
     filter_check,
 };
 
-#[derive(Debug, Enum, EnumIter, Eq, PartialEq, Copy, Clone, Ord, PartialOrd)]
+//#[derive(Debug, Enum, EnumIter, Eq, PartialEq, Copy, Clone, Ord, PartialOrd)]
```
[the remark this answers is not in the document]
Comment (tobz) [reply src/api/schema/components/source.rs:16]: Nope, definitely a spot I forgot to clean up. Good catch.

File: src/config/loading/loader.rs [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
-use strum_macros::EnumIter;
 use toml::value::{Table, Value};
 
 use super::{component_name, open_file, read_dir, Format};
 use crate::config::format;
 
 /// Provides a hint to the loading system of the type of components that should be found
 /// when traversing an explicitly named directory.
-#[derive(Debug, Copy, Clone, EnumIter)]
+//#[derive(Debug, Copy, Clone, EnumIter)]
```
[the remark this answers is not in the document]
Comment (tobz) [reply src/config/loading/loader.rs:11]: Nope, definitely a spot I forgot to clean up. Good catch.

[the remark this answers is not in the document]
Comment (tobz) [reply src/config/loading/loader.rs]: Fixed in [7f41ad6](https://github.com/vectordotdev/vector/pull/13172/commits/7f41ad679d0e2719ff0dadfa526d98f4b78e4dfa).

File: src/api/schema/components/source.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobz) [reply src/api/schema/components/source.rs:16]: Fixed in [7f41ad6](https://github.com/vectordotdev/vector/pull/13172/commits/7f41ad679d0e2719ff0dadfa526d98f4b78e4dfa).

**Review by spencergilbert — APPROVED** [review]
I'm confused as to why stackdriver logs and metrics were using different auth libs 🤦

**Comment by tobz** [comment]
I think it was a contributor vs team addition thingamajig... but 🤷🏻.

**Review by bruceg — APPROVED** [review]

**Pull request merged**